test/Rings-conformance-tests.jl

```diff
@@ -386,6 +511,14 @@ function test_EuclideanRing_interface(R::AbstractAlgebra.Ring; reps = 20)
          @test d == gcd(f, g)
          @test d == s*f + t*g
          @test gcdinv(f, g) == (d, s)
+
+         if !is_zero(f) && !is_zero(g) && !is_zero(m)
```

Comment:
I think the need for `f` and `g` to be non-zero would go if you adopt my proposal above.